crates/ty_python_semantic/src/types.rs

```diff
@@ -10950,31 +10977,42 @@ impl get_size2::GetSize for CallableType<'_> {}
 impl<'db> CallableType<'db> {
     /// Create a callable type with a single non-overloaded signature.
     pub(crate) fn single(db: &'db dyn Db, signature: Signature<'db>) -> Type<'db> {
```

Comment:
I sort-of feel like `CallableType::single` should return `CallableType`, and we should have a `Type::simple_callable` method that wraps that in a `Type`, similar to these helper methods we have elsewhere:

https://github.com/astral-sh/ruff/blob/e642874cf1a80598242e3d2becf684ed8d4c4597/crates/ty_python_semantic/src/types.rs#L1276-L1286

but this issue predates your PR, so don't worry about fixing it here :-)